### PR TITLE
Added LiNA/MDLiNA classes w.r.t the paper "Causal Discovery with Multi-Domain LiNGAM for Latent Factors". 

### PR DESCRIPTION
**Pull request by YanaZeng**
Hi everyone,

I am Yan Zeng (Yana). I have added the LiNA/MDLiNA methods w.r.t the paper "Causal Discovery with Multi-Domain LiNGAM for Latent Factors", which are based on the lingam code [v1.5.4](https://github.com/cdt15/lingam/releases/tag/v1.5.4). 

Please feel free to check it. This is my first time using such a "pull request" function, so if there are any problems, please let me know. ;)

Best wishes,
Yana

### CODE REVIEW

**Comment by ikeuchi-screen** [comment]
Hi, @YanaZeng. Thank you for adding the implementation of the new algorithm.
It may take some time to check due to the many great additions, so I hope you can wait a little while.

First, I checked the file name included in the pull request.
I think the following 6 files are configuration files, is it possible to delete them?
* .idea/deployment.xml
* .idea/inspectionProfiles/profiles_settings.xml
* .idea/lingam.iml
* .idea/modules.xml
* .idea/vcs.xml
* .idea/workspace.xml

File: lingam/lina.py [hunk 1]
```diff
@@ -0,0 +1,629 @@
+"""
+Python implementation of the LiNGAM algorithms.
+The LiNGAM Project: https://sites.google.com/site/sshimizu06/lingam
+"""
+
+import numpy as np
+import scipy.optimize as sopt
+from scipy.special import expit as sigmoid
+from factor_analyzer import (ConfirmatoryFactorAnalyzer, ModelSpecificationParser)
+import random
```
Comment (ikeuchi-screen) [inline lingam/lina.py:10]: 'random' imported but unused

File: lingam/utils/__init__.py [hunk 2]
```diff
@@ -18,9 +23,244 @@
...
 
+def set_random_seed(seed):
+    random.seed(seed)
+    np.random.seed(seed)
+
+
+def simulate_linear_sem(W, n, sem_type, noise_scale=1.0):
+    """Simulate samples from linear SEM with specified type of noise.
+
+    Args:
```
Comment (ikeuchi-screen) [inline lingam/utils/__init__.py:42]: The docstring of the method added to `__init__.py` is not in Numpy style.

**Review by ikeuchi-screen — CHANGES_REQUESTED** [review]
@YanaZeng, I left some comments. Please take a look.

Also, please add igraph and factor_analyzer to your requirements.txt file, because I got the following error when I ran the test:
* ModuleNotFoundError: No module named 'igraph'
* ModuleNotFoundError: No module named 'factor_analyzer'

File: lingam/lina.py (hunk 1 above)
Comment (YanaZeng) [reply lingam/lina.py:10]: Thanks for your carefulness! Please delete "import random".

File: lingam/utils/__init__.py (hunk 2 above)
Comment (YanaZeng) [reply lingam/utils/__init__.py]: I have uploaded the newly-revised __init__.py in https://github.com/YanaZeng/lingam/tree/add-lina-algor/lingam/utils. Please see it.

**Comment by YanaZeng** [comment]
Hi, @ikeuchi-screen. I am very sorry for the inconvenience caused you. 

I have added the igraph and factor_analyzer to the requirements.txt file as well as added some descriptions on the LiNA codes in the README.md file [https://github.com/YanaZeng/lingam/tree/add-lina-algor]. For the other comments, I have replied to them as above. Please feel free to see them.

**Review by ikeuchi-screen — APPROVED** [review]

**Pull request merged**